contract_comm/currency/currency.go

```diff
@@ -94,97 +94,127 @@ const (
 )
 
 var (
-	cgExchangeRateNum = big.NewInt(1)
-	cgExchangeRateDen = big.NewInt(1)
-
 	medianRateFuncABI, _   = abi.JSON(strings.NewReader(medianRateABI))
 	balanceOfFuncABI, _    = abi.JSON(strings.NewReader(balanceOfABI))
 	getWhitelistFuncABI, _ = abi.JSON(strings.NewReader(getWhitelistABI))
 )
 
-type exchangeRate struct {
-	Numerator   *big.Int
-	Denominator *big.Int
+// ExchangeRate represent the exchangeRate for the pair (base, token)
+// Follows the equation: 1 base * ExchangeRate = X token
+type ExchangeRate struct {
```

Comment:
Not actually used outside this package.

Reply:
hehe... we are going to have this fight again...

To me `ExchangeRate`, `MustNewExchangeRate()` and `GetExchangeRate()` makes perfect sense to be public on a module that is about currencies and exchanges rates.

It's true, no other module is currently using it. But from a "internal module perspective", is IMO are perfect candidates for publishing.

I intentially wanted to expose ExchangeRate, and not expose `numerator` and `denmonitor` since i don't think the caller should be aware of that implementation detail